### ISSUE TITLE
Marker CLI changes

### ISSUE THREAD

**Issue by aeshky** [RasaHQ/rasa#10174]

**Comment by usc-m** [comment]
2/3 and are trivial and good ideas so that sounds good.

1 is a bit fiddly because with the current design we have a tradeoff between exclusivity (making sure users can't input invalid commands) and prettiness here. The way `argparse` works, we can't have mutually exclusive groups of arguments contingent on the `strategy` without using sub-parsers. The way sub-parsers work they will show their own help independently when you call `--help` on them directly. One way around this is to add all the arguments to the parser a level up (in this case `markers`), but then we lose the exclusivity property and have to implement validation to make sure commands are still valid - this would achieve what you're describing.

However, I think a better idea is to actually push the common arguments out to the individual strategies, so that `--help` on `evaluate markers` would show you the list of strategies, and then you would pick one and then call `--help` on that to see *all* of the arguments. This doesn't create any extra validation hassle, means help only shows exactly what it needs to, and makes the command flow a bit better (`rasa evaluate markers first_n 100 output.csv --config blah.yaml`). The main extra hassle is making sure all of the duplicated arguments don't interfere etc (which is more QA related than writing extra validation code)

Another suggestion I saw was to change `--config` to `--markers` by @koaning since that clarifies that we're not talking about the model config, but the markers config - we could also do that here too.

**Comment by aeshky** [comment]
`rasa evaluate markers first_n 100 output.csv --config blah.yaml` looks good 👍 

I first attempted `rasa evaluate markers first_n --help`, then realised I had to insert the file, so I prefer the design you're suggesting.

As for `--config`, I prefer it over `--markers`. I worry we might call too many things `markers` (e.g., the marker configurations, vs marker output).

**Comment by koaning** [comment]
There's a separate issue for `--help` commands in general here that may be relevant to be aware of: https://github.com/RasaHQ/rasa/issues/10103.
